cpp_api/api.h

```diff
@@ -0,0 +1,40 @@
+#include <memory>
+#include <vector>
+
+// Forward declaration of API dependencies
+class goto_modelt;
+class message_handlert;
+class optionst;
+
+#include "options.h"
+
+// An object in the pattern of Session Facade - owning all of the memory
+// the API is using and being responsible for the management of that.
+struct cbmc_api
+{
+  // Initialising constructor
+  explicit cbmc_api();
+  cbmc_api(const api_optionst &options);
+  ~cbmc_api();
```

Comment:
I suggest that we put a comment on this destructor in the header file, noting that it is defaulted in the `.cpp` file for compatibility with implementing the "pointer to implementation" pattern with `unique_ptr`. This saves checking the `.cpp` file to work this out, as the prospect of a hand-written destructor implementation can be concerning.